### ISSUE TITLE
[a11y]: Screen reader misreads datagrid table headers

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#5682]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

v2.44.0-rc.1

### Automated testing tool and ruleset

manual

### Assistive technology

JAWS

### Description

Reported by Raghu during screen reader testing.

The screen reader incorrectly announces table header labels when navigating the Datagrid. This disrupts the user experience for screen reader users as they cannot associate data cells with their corresponding headers.

**Example:**
Once user navigates to data cell in column 2, the screen reader announces “Column 3 header” instead of “Column 2 header” 

**Recommended fix**
- Ensure the reading order of the table structure aligns with the visual layout.
- Screen reader should read the corresponding column headers, while navigating through each column

**User impact**
Screen reader users cannot understand the relationship between data and headers, hindering comprehension of the table content. 

### WCAG 2.1 Violation

1.3.1. Info & Relationships

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-datagrid--radio-select&globals=viewport:basic

### Steps to reproduce

Documented in the following Datagrid stories:
Radio select, selectable row, hide select all, actions dropdown, batch actions, disable select row, frozen columns

1. Using JAWS screen reader, navigate through column headers.
2. Screen reader will announce the incorrect header.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
